docs/UnitTestBarAndStandards.md

```diff
@@ -0,0 +1,23 @@
+<!-- Copyright (c) Microsoft Corporation. All rights reserved.
+     Licensed under the MIT License. -->
+     
+     ## Test requirements
+* All new non-UI code must have unit tests.
+* Unit tests need to accompany the code in the pull request. This provides implicit documentation of the expected behavior, and ensures that the behavior is being satisfied. In an exceptional circumstance, a PR might be approved without its unit tests, but that is not the normal expectation.
+* All unit tests should be passing locally before the PR is created.
+* Unit tests must conform to the standards listed below.
+
+## Standards
+* The test project is appropriately named (tests for the `Foo.Bar.dll` are in `Foo.BarTests.dll`)
+* The test namespace is appropriately named (the top level namespace matches the assembly name, then the hierarchy in the assembly is preserved. So if `Foo.Bar.dll` contains a namespace named `Widget`, the tests will be in the `Foo.BarTests.Widget` namespace)
+* The test class is appropriately named (tests for the `ExplodingWidget` class will be in the `ExplodingWidgetTests` class)
```

Comment:
>ExplodingWidgetTests [](start = 95, length = 20)

Several classes currently follow the *UnitTests convention instead of the *Tests convention. I have strong preference for the former